wcfsetup/install/files/lib/system/html/output/node/HtmlOutputNodeNormalizer.class.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace wcf\system\html\output\node;
+
+use wcf\util\DOMUtil;
+
+/**
+ * Normalizes HTML generated by earlier version of WoltLab Suite.
+ *
+ * @author Alexander Ebert
+ * @copyright 2001-2023 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.0
+ */
+final class HtmlOutputNodeNormalizer
+{
+    public function __construct(private readonly \DOMXPath $xpath)
```

Comment:
This parameter should be a parameter of `normalize()` and then passed along. This removes the implicit dependency on `$this->xpath` in some methods (not all need access to the xpath!) and would make objects of the normalizer reusable for normalizing multiple messages.